k8s/src/main/scala/io/buoyant/k8s/resources.scala

```diff
@@ -174,19 +189,22 @@ private[k8s] class NsListResource[O <: KubeObject: TypeReference, W <: Watch[O]:
 }
 
 private[k8s] class NsObjectResource[O <: KubeObject: TypeReference, W <: Watch[O]: TypeReference](
-  val client: Client,
-  listPath: String,
+  parent: Resource,
   objectName: String,
+  maybeListName: Option[String] = None,
   protected val backoffs: Stream[Duration] = Watchable.DefaultBackoff,
   protected val stats: StatsReceiver = DefaultStatsReceiver
-)(implicit od: ObjectDescriptor[O, W]) extends Resource {
-  private[this] val path = s"$listPath/$objectName"
-
-  def get: Future[O] = {
-    val req = Api.mkreq(http.Method.Get, path, None)
-    Trace.letClear(client(req)).flatMap { rsp =>
-      Api.parse[O](rsp)
-    }
+)(implicit od: ObjectDescriptor[O, W])
+  extends Watchable[O, W, O]
+  with Resource {
+
+  override val client: Client = parent.client
+  override val (path, watchPath) = {
+    // the value of listName can be dropped when we're done with it,
+    // so allocate it in a scope.
```

Comment:
This is not in a hot path so the goal should be to maximize clarity not minimize allocations.